src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      (a + b <= c || a + c <= b || b + c <= a)
+      || (a <= 0 || b <= 0 || c <= 0)
+    ) {
+      throw new Error('Wrong figure parameters');
```

Comment:
i suppose would be better if you create 2 different errors. 'cause user does not understand what mistake he made